### PR TITLE
Add new ProfilesContainer component for HashrateChallenge variants

### LINKED ISSUE #503: HashrateChallenge component
Chapter 3 primarily consists of hashrate battles between 2 or more parties. To simplify the creation of the different challenges for Chapter 3 we should create a reusable `<HashrateChallenge>` component. This is what it would look like when rendered:

#### 1 vs. 1
image

#### 4 vs. 1 (Idle)
image

#### 4 vs. 1 (In progress)
image

#### 4 vs. 1 (Completed)
image

As you can see from the designs the component should:
- Support a dynamic amount of participants.
- Have different states (Idle, In progress, Completed).

When building this component we should reuse the components we created in:
https://github.com/saving-satoshi/saving-satoshi/issues/444
https://github.com/saving-satoshi/saving-satoshi/issues/448
https://github.com/saving-satoshi/saving-satoshi/issues/446

### PR DESCRIPTION
**Pull request by benalleng**
prelminary work for #503 

creates new Scrolling Profile container and 2 new lesson templates to demonstrate the functionality, does not add the complete HashrateChallenge

[Solo Preview](https://saving-satoshi-git-fork-benalleng-issue503-savingsatoshi.vercel.app/en/chapters/chapter-3/solo-1?dev=true)

[Pool Preview](https://saving-satoshi-git-fork-benalleng-issue503-savingsatoshi.vercel.app/en/chapters/chapter-3/pool-1?dev=true)

### CODE REVIEW

**Comment by ecurrencyhodler** [comment]
Okay so I tested this on desktop and mobile for chrome/firefox and apple devices.

Scrolling is smooth on desktop and mobile. 

https://github.com/saving-satoshi/saving-satoshi/assets/32662508/f90a2629-0546-489e-8778-5e54ce1d2b5e


But it looks like the profiles are more squarish than rectangular. Will that create an issue with profile images?
IMG_2854

**Comment by benalleng** [comment]
the width of the profiles are just hard coded shapes at the moment, once they are the actual profiles their shapes may vary _though i did try to match their sizes to the figma_

**Comment by ecurrencyhodler** [comment]
Okay then tACK

File: ui/common/ProfilesContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import { ReactNode, useRef, useEffect, useState } from 'react'
+
+export default function ProfilesContainer({
+  children,
+  className,
+  profiles,
+  verticalProfiles,
+}: {
+  children?: ReactNode
+  className?: string
```
Comment (vinayaksh42) [inline ui/common/ProfilesContainer.tsx:13]: Is this prop being used somewhere within the component right now?

**Review by vinayaksh42 — COMMENTED** [review]
the code looks good to me, just one question regarding one of the prop

File: content/lessons/chapter-3/index.ts [hunk 2]
```diff
@@ -1,11 +1,13 @@
 import * as Intro1 from './intro-1'
 import * as Intro2 from './intro-2'
 import * as SOL1 from './solo-1'
+import * as POL1 from './pool-1'
```
Comment (afterburn) [inline content/lessons/chapter-3/index.ts:4]: Let's use `Solo1` / `Pool1` to match the existing naming convention.

**Review by afterburn — CHANGES_REQUESTED** [review]
Thanks @benalleng, small nitpick from me and agree with @vinayaksh42 that we should probably use that className somewhere.

**Pull request merged**